src/vs/editor/common/model.ts

```diff
@@ -589,6 +589,12 @@ export interface ITextModel {
 	 */
 	readonly isForSimpleWidget: boolean;
 
+
+	/** // {{SQL CARBON EDIT}}
+	 * The URI of the notebook that contains this text model as part of a notebook cell.
+	 */
+	notebookUri?: URI;
```

Comment:
Would you be able to instead make an `INotebookTextModel extends ITextModel` (put in src/sql/editor/common/model.ts) and then include the notebookUri? Having this field for *every* text model is not something we should have.